topdown/bytes_test.go

```diff
@@ -0,0 +1,127 @@
+package topdown
+
+import (
+	"github.com/open-policy-agent/opa/ast"
+	"testing"
+)
+
+func TestNumBytes(t *testing.T) {
+	t.Run("SuccessfulParse", func(t *testing.T) {
+		tests := []struct {
+			note, rule string
+			expected   int
+		}{
+			{"zero", `0`, 0},
+			{"raw number", `12345`, 12345},
+			{"7k", `7k`, 7 * kb},
+			{"8 kilobytes", `8k`, 8 * kb},
+			{"10 kilobytes uppercase", `10KB`, 10 * kb},
+			{"10 kilobytes lowercase", `10kb`, 10 * kb},
+			{"10 kilobytes mixed case", `10Kb`, 10 * kb},
+			{"200 megabytes as m", `200m`, 200 * mb},
+			{"200 megabytes as mb", `200mb`, 200 * mb},
+			{"37 gigabytes", `37g`, 37 * gb},
+			{"3 terabytes", `3t`, 3 * tb},
+			{"Float", `10.2`, 10},
```

Comment:
We just ignore decimal places right now, right? So `0.9T` == 0 bytes?

Reply:
Yeah, this is tricky and one of the main reasons why I originally marked this as a WIP. I legitimately don't know what the preferred path is here. On the one hand, converting "9.7mb" to 9.7 * 1024 * 1024 (or 9.7 * 1000 * 1000 😉) makes this function more *useful*, but I'm not sure if people actually do that in the wild or expect to be able to. Your example illustrates drives home that Least Surprise probably dictates that floats should be respected. Happy to defer to others on this.

Reply:
I think we should support floats if it doesn't cause too many issues.

If we switch to parsing the number string it is pretty easy to just allow a single `.` character too and then dump the whole slice into a big float.